### PR TITLE
Compare length of resetKeys when checking if resetKeys has changed

### PR DESCRIPTION
**Pull request by mmiller42**
**What**:

When determining whether to reset state/call `onResetKeysChange` callback, also checks if the length of the `resetKeys` dependency array changes.

**Why**:

Edge case where `resetKeys` array is computed dynamically and its length could change.

**How**:

Added a length comparison check to `changedArray`.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/__tests__/index.js [hunk 1]
```diff
@@ -316,6 +321,39 @@ test('supports automatic reset of error boundary when resetKeys change', () => {
...
+  expect(console.error).toHaveBeenCalledTimes(2)
+  screen.getByRole('alert')
+  console.error.mockClear()
+  userEvent.click(screen.getByText('toggle extra resetKey'))
+  expect(handleReset).not.toHaveBeenCalled()
+  expect(handleResetKeysChange).toHaveBeenCalledTimes(1)
+  expect(handleResetKeysChange).toHaveBeenCalledWith([false, true], [false])
+  handleResetKeysChange.mockClear()
+  expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+  expect(console.error).not.toHaveBeenCalled()
```
Comment (kentcdodds) [inline src/__tests__/index.js:356]: This is a pretty big block of code with no breaks. Could you add some logical breaks and comments through this to indicate what's happening?

**Comment by kentcdodds** [comment]
@all-contributors please add @mmiller42 for code and tests
